earthfile2llb/converter.go

```diff
@@ -477,14 +478,21 @@ func (c *Converter) Volume(ctx context.Context, volumes []string) {
 }
 
 // Env applies the ENV command.
-func (c *Converter) Env(ctx context.Context, envKey string, envValue string) {
+func (c *Converter) Env(ctx context.Context, envKey string, envValue string) error {
 	envValue = c.expandArgs(envValue)
+	invalidEnvKey, err := regexp.MatchString(`[${}()]`, envKey)
+	logging.GetLogger(ctx).With("env-key", envKey).With("env-key-value", invalidEnvKey).Info("ENV Key Value")
+	if invalidEnvKey || err != nil {
+		logging.GetLogger(ctx).Info("Failed to apply ENV")
+		return fmt.Errorf("Invalid env key definition %s", envKey)
+	}
```

Comment:
Side note: No need to do logging - just make sure `l.err` is set after you move to listener.